strings_lab.js

```diff
@@ -0,0 +1,220 @@
+}
+
+
+
+
+
+// 6. Create a `verbing` code block. It should take a single string variable.
+// If its length is at least 3, it should add 'ing' to its end, unless it
+// already ends in 'ing', in which case it should add 'ly' instead. If the
+// string length is less than 3, it should leave it unchanged.
+
+// For example:
+//
+// ```js
+// verbing('box')
+// // => 'boxing'
+// verbing('train')
+// // => 'training'
+// verbing('swimming')
+// // =>  'swimmingly'
+// verbing('go')
+// // =>  'go'
+// ```
+//
+
+
+
+let verb = "go";
+let verbing = verb.slice(verb.length - 3, verb.length)
+
+if (verbing === "ing") {
+  console.log(verb + "ly");
+} else if (verb.length > 3) {
+  console.log(verb + "ing");
+} else {
+  console.log(verb);
+}
+
+
+// 7. Create a `withoutLast` code block that takes a single string variable,
+// and returns a copy of the string without the last word.
+//
+
+
+let word = "welcome";
+let withoutLast = word.slice(0, word.length - 1);
+
+ console.log(withoutLast);
+
+
+
+// ## Bonus Questions
+//
+// 1. Write a code block that adds a "#" to a string if the string has an odd
+// number of characters, otherwise leave it as is.
+// print/log the result
+//
+//     Examples:
+//
+//   ```js
+//   let name = 'taq'  // should log 'taq#'
+//   let name2 ='john' // should log 'john'
+//   ```
+
+
+let name = "john";
+
+ if (name.length % 2 == 0){
+  console.log(name);
+}else{
+  console.log(name + "#");
+}
+
+
+// ___
+//
+// 2. Write a code block that prints/logs the 3rd to last characters of a string.
+// If the string is not at least 3 characters long then log an error message.
+// <details>
+//     <summary>
+//         Hint
+//     </summary>
+//     It should work for strings of any length.
+// </details>
+//
+// Examples, for variables:
+// ```js
+// let name1 = 'Taq' // Should log 'Taq'
+// let sentence = 'He is Taq'// Should log 'Taq'
+// ```
+
+let string= "hello";
+
+ if (string.length >= 3){
+  console.log(string.slice(string.length - 3));
+}else{
+  console.log("This string is too short");
+}
+
+
+
+
+// ___
+//
+// 3. Write a code block that prints/logs whether a string `x` starts with a vowel.
+// <details>
+//   <summary>
+//     Extra
+//   </summary>
+//
+//   Make it [case insensitive](https://en.wikipedia.org/wiki/Case_sensitivity)
+//
+// </details>
+//
+
+//
+let x = "Alphabet";
+let vowel = "AEIOUaeiou";
+ if (vowel.includes(x[0])) {
+  console.log("This word begins with a vowel");
+}else {
+  console.log("This word begins with a consonent");
+}
+
+
+// ___
+//
+// 4. Write a code block that given two strings, `str1` and `str2`, prints/logs
+// a new string with `str2` mashed into the middle of `str1`
+//
+// Examples:
+//
+//   - For `str1` = 'help' and `str2` = 'me'. Your program should log `'hemelp'`
+//
+//    - For `str1` = 'hello' and `str2` = 'world'. Your program should log `'helworldlo'`
+//
+
+
+let str1 = "hello"
+let str2 = "world"
+let midstr1 = Math.ceil(str1.length/2);
```

Comment:
Setting midstr1 is definitely a good idea to solve problems like this. Well done!